### PR TITLE
Bug 1173522 - Tie the factory reset toggle to the warning dialogs r=eragonj

### PR DESCRIPTION
**Pull request by crh0716**
https://bugzilla.mozilla.org/show_bug.cgi?id=1173522

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=33cb788630489d36e1acb1e74c955ac6687687f2

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4f98dd55987f5cfecb13c26498ecbb1db8e548e2

File: apps/settings/js/modules/dialog_panel.js [hunk 1]
```diff
@@ -3,17 +3,48 @@ define(function(require) {
...
+    };
+
+    // Callee can use `this.cancel()` to programmatically cancel()
+    // this dialog.
+    panelOptions.cancel = function() {
+      cancelFunction();
+    };
+
     return panel;
   };
```
Comment (EragonJ) [inline apps/settings/js/modules/dialog_panel.js]: Why this code change is so familiar XDD

Comment (crh0716) [inline apps/settings/js/modules/dialog_panel.js]: Ha, I couldn't wait to use it so I stole the code from your patch.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c00fca737d4a960d29aa1138727d9819805ea3da

**Pull request merged**